providers/provider.go

```diff
@@ -187,6 +187,11 @@ type PlanResourceChangeRequest struct {
 	// changes required to create the proposed final state.
 	ProposedNewState cty.Value
 
+	// Config is the resource configuration, before being merged
+	// with the PriorState. Config is supplied for reference, but Provider
+	// implementations should prefer the ProposedNewState in most circumstances.
```

Comment:
Perhaps worth stating explicitly the usage we had in mind here: any value not explicitly set in the configuration will be null.